pkg/vpcmodel/semanticDiffSubnets.go

```diff
@@ -213,10 +188,325 @@ func diffDescription(diff DiffType) string {
+	err = nil
+	alignedConnectivity = map[EndpointElem]map[EndpointElem]*common.ConnectionSet{}
+	for src, endpointConns := range *subnetConnectivity {
+		for dst, conns := range endpointConns {
+			if conns.IsEmpty() {
+				continue
+			}
+			if _, ok := alignedConnectivity[src]; !ok {
+				alignedConnectivity[src] = map[EndpointElem]*common.ConnectionSet{}
+			}
+			// the resizing element is not external - copy as is
+			if (resizeSrc && !src.IsExternal()) || (!resizeSrc && !dst.IsExternal()) {
+				alignedConnectivity[src][dst] = conns
+				continue
+			}
+			// the resizing element is external - go over all ipBlock and allocates the connection
+			// if the ipBlock is contained in the original src/dst
+			var origIPBlock *common.IPBlock
+			if resizeSrc {
+				origIPBlock, err = externalNodeToIPBlock(src.(Node))
+			} else {
+				origIPBlock, err = externalNodeToIPBlock(dst.(Node))
+			}
+			if err != nil {
+				return nil, err
+			}
+			err = config.addIPBlockToConnectivityMap(disjointIPblocks, origIPBlock, alignedConnectivity, src, dst, conns, resizeSrc)
+		}
+	}
+	return alignedConnectivity, err
+}
+
+func (c *VPCConfig) addIPBlockToConnectivityMap(disjointIPblocks []*common.IPBlock,
+	origIPBlock *common.IPBlock, alignedConnectivity map[EndpointElem]map[EndpointElem]*common.ConnectionSet,
+	src, dst EndpointElem, conns *common.ConnectionSet, resizeSrc bool) error {
+	origBlockContainsIPBlocks := false
+	for _, ipBlock := range disjointIPblocks {
+		// 1. get ipBlock of resized index (src/dst)
+		if !ipBlock.ContainedIn(origIPBlock) { // ipBlock not relevant here
+			continue
+		}
+		origBlockContainsIPBlocks = true
+		cidrList := ipBlock.ToCidrList()
+		for _, cidr := range cidrList {
+			nodeOfCidr, err := findNodeWithCidr(c.Nodes, cidr)
+			if err != nil {
+				return err
+			}
+			if resizeSrc {
+				if _, ok := alignedConnectivity[nodeOfCidr]; !ok {
+					alignedConnectivity[nodeOfCidr] = map[EndpointElem]*common.ConnectionSet{}
+				}
+				alignedConnectivity[nodeOfCidr][dst] = conns
+			} else {
+				alignedConnectivity[src][nodeOfCidr] = conns
+			}
+		}
+	}
+	if !origBlockContainsIPBlocks { // keep as origin
+		alignedConnectivity[src][dst] = conns
+	}
+	return nil
+}
+
+// gets node with given cidr
+func findNodeWithCidr(configNodes []Node, cidr string) (Node, error) {
+	for _, node := range configNodes {
+		if node.Cidr() == cidr {
+			return node, nil
+		}
+	}
+	return nil, fmt.Errorf("%s", fmt.Sprintf("A node with cidr %v not found in conf", cidr)) // should not get here
+}
+
+// get a list of IPBlocks of the src and dst of the connections
+func (subnetConnectivity SubnetConnectivityMap) getIPBlocksList() (ipbList []*common.IPBlock,
+	myErr error) {
+	ipbList = []*common.IPBlock{}
+	myErr = nil
+	for src, endpointConns := range subnetConnectivity {
+		for dst, conns := range endpointConns {
+			if conns.IsEmpty() {
+				continue
+			}
+			if src.IsExternal() {
+				ipBlock, err := externalNodeToIPBlock(src.(Node))
```

Comment:
`src` or `dst` are of type `EndpointElem`, which can also be external as a `groupedExternalNodes`.. (though not expected to happen because it is not yet grouped)..
The question is, if this is prior grouping, why is the connectivityMap (type `SubnetConnectivityMap`) in terms of `EndpointElem` rather than `VPCResourceIntf`? I find it a bit confusing..

Also, there are more places in the code where type conversions are not checked -- please check conversions are ok there as well.

Reply:
errors handled; [EndpointElem  -> VPCResourceIntf ](url)will be done as a preliminary step to https://github.com/np-guard/vpc-network-config-analyzer/issues/211